### PR TITLE
BUG: Fix bug with plot_epochs_image

### LINKED ISSUE #11339: Wrong adjacency when calculated on lateral channel subset
### Description of the problem

When calculating channel adjacency for a (lateral) subset of EEG channels (e.g., only left channels), I seem to get a slightly wrong adjacency matrix that also exhibits adjacency of clearly non-neighbouring channels (in my case: `F1`and `P1` and also `FT9` and `TP9`). 
When I use the same procedure on the full montage, the same channels are not denoted as adjacent. 

I did not dig very deep as I'm only superficially familiar with calculating adjacency/triangulation, and I might very well be doing something wrong or this is expected behavior. 
Thanks for clarification!

### Steps to reproduce

```Python
import mne
import matplotlib.pyplot as plt

easycap_montage = mne.channels.make_standard_montage('easycap-M1')
info_full = mne.create_info(ch_names=easycap_montage.ch_names,
                            sfreq=1,
                            ch_types='eeg')
info_full.set_montage(easycap_montage)
ch_sel = mne.channels.make_1020_channel_selections(info_full)
chs_left = [info_full.ch_names[i] for i in ch_sel['Left']]
info_left = mne.create_info(ch_names=chs_left,
                            sfreq=1,
                            ch_types='eeg')
info_left.set_montage(easycap_montage)
chan_adj, ch_n = mne.channels.find_ch_adjacency(info_left, 'eeg')
fig = mne.viz.plot_sensors(info_left, show_names=True)
# looks ok and, importantly, P1 and F1 are no neighbours

fig, ax = plt.subplots(figsize=(16,16))
ax.spy(chan_adj)
ax.set_xticks(range(len(ch_n)))
ax.set_xticklabels(ch_n)
ax.set_yticks(range(len(ch_n)))
ax.set_yticklabels(ch_n)
plt.show()

idx_f1 = info_left.ch_names.index('F1')
idx_p1 = info_left.ch_names.index('P1')
print("P1 and F1 adjacency (left sensors): " + str(chan_adj[idx_f1, idx_p1]))

# whereas:
chan_adj_full, ch_n_full = mne.channels.find_ch_adjacency(info_full, 'eeg')
idx_f1 = info_full.ch_names.index('F1')
idx_p1 = info_full.ch_names.index('P1')
print("P1 and F1 adjacency (all sensors): " + str(chan_adj_full[idx_f1, idx_p1]))
```

### Link to data

_No response_

### Expected results

Channels which are quite far apart (like `P1` and `F1`) should not be declared as "adjacent". 
Also, should the adjacency of such an isolated subset of channels not be a subset/"sub-matrix" of the adjacency matrix on the full channel set? 

### Actual results

see output of code snippet

### Additional information

Platform:         Linux-5.3.18-150300.59.87-default-x86_64-with-glibc2.31
Python:           3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:35:26) [GCC 10.4.0]
Executable:       /u/fklotzsche/conda-envs/vr2fem/bin/python
CPU:              x86_64: 144 cores
Memory:           503.2 GB

mne:              1.1.1
numpy:            1.23.3 {blas=mkl_rt, lapack=mkl_rt}
scipy:            1.9.1
matplotlib:       3.5.2 {backend=module://matplotlib_inline.backend_inline}

sklearn:          1.1.2
numba:            Not found
nibabel:          4.0.2
nilearn:          0.9.2
dipy:             1.5.0
cupy:             Not found
pandas:           1.4.4
pyvista:          0.36.1 {OpenGL 3.1 Mesa 20.2.4 via llvmpipe (LLVM 11.0.1, 256 bits)}
pyvistaqt:        0.9.0
ipyvtklink:       0.2.3
vtk:              9.1.0
qtpy:             2.2.0 {PyQt5=5.15.2}
ipympl:           Not found
pyqtgraph:        0.12.3
pooch:            v1.6.0

mne_bids:         Not found
mne_nirs:         Not found
mne_features:     Not found
mne_qt_browser:   0.4.0
mne_connectivity: Not found
mne_icalabel:     0.4

### PR DESCRIPTION
**Pull request by larsoner**
Closes #11339

### CODE REVIEW

File: mne/channels/channels.py [hunk 1]
```diff
@@ -1600,6 +1600,14 @@ def find_ch_adjacency(info, ch_type):
     you want to load a template for a given montage use
     :func:`read_ch_adjacency` directly.
 
+    .. warning::
+        If Delaunay triangulation is used to calculate the adjacency matrix it
+        may yield partially unexpected results (e.g., include unwanted edges
+        between non-adjacent sensors). Therefore, it is recommended to check
+        (and, if necessary, manually modify) the result by inspecting it
+        via :func:`mne.viz.plot_ch_adjacency`.
+        Delete this line @eort...
```
Comment (drammock) [inline mne/channels/channels.py:1609]: should be @eioe not @eort

Comment (larsoner) [inline mne/channels/channels.py:1609]: @eort can you reply to this comment on GH using the document plus/minus suggest-change feature suggesting to delete this line? Then I can commit it, and you should get co-author credit for this change

It's a hack but it was the easiest way I saw to get you credit in the commit history

Comment (drammock) [reply mne/channels/channels.py:1609]: FYI you can also manually type
```
Co-authored-by: Name Surname <email@addre.ss>
```
on line 3 of a multi-line commit message. But you have to get the syntax just right and use a name/email just as they're spelled in our mailmap.  So it's not actually easier :)

Comment (eioe) [reply mne/channels/channels.py:1609]:
```suggestion
```
Like this? 
thanks for your efforts. 🙏Would not have been necessary.

Comment (larsoner) [reply mne/channels/channels.py]: Perfect, thanks :+1:

Comment (eioe) [reply mne/channels/channels.py]: > FYI you can also manually type
> 
> ```
> Co-authored-by: Name Surname <email@addre.ss>
> ```
> 
> on line 3 of a multi-line commit message. But you have to get the syntax just right and use a name/email just as they're spelled in our mailmap. So it's not actually easier :)

I love this comment on multiple levels. 😂

**Pull request merged**